appinventor/components/src/com/google/appinventor/components/runtime/ChartDataBase.java

```diff
@@ -0,0 +1,141 @@
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleFunction;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.runtime.util.YailList;
+
+@SimpleObject
+public abstract class ChartDataBase implements Component {
+    protected Chart container;
+    protected ChartDataModel chartDataModel;
+
+    private String label;
+    private int color;
+
+    /**
+     * Creates a new Chart Data component.
+     */
+    protected ChartDataBase(Chart chartContainer) {
+        this.container = chartContainer;
+        chartContainer.addDataComponent(this);
+        initChartData();
+    }
+
+    /**
+     * Returns the data series color as an alpha-red-green-blue integer.
+     *
+     * @return  background RGB color with alpha
+     */
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public int Color() {
+        return color;
+    }
+
+    /**
+     * Specifies the data series color as an alpha-red-green-blue integer.
+     *
+     * @param argb  background RGB color with alpha
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR,
+            defaultValue = Component.DEFAULT_VALUE_COLOR_BLACK)
+    @SimpleProperty
+    public void Color(int argb) {
+        color = argb;
+        chartDataModel.setColor(color);
+        refreshChart();
+    }
+
+    /**
+     * Returns the label text of the data series.
+     *
+     * @return  label text
+     */
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public String Label() {
+        return label;
+    }
+
+    /**
+     * Specifies the text for the data series label.
+     *
+     * @param text  label text
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+            defaultValue = "")
+    @SimpleProperty
+    public void Label(String text) {
+        this.label = text;
+        chartDataModel.setLabel(text);
+        refreshChart();
+    }
+
+    /**
+     * Specifies the elements of the entries that the Chart should have.
+     * @param elements  Comma-separated values of Chart entries alternating between x and y values.
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING, defaultValue = "")
+    @SimpleProperty(description="To be done (non-functional for now)",  category = PropertyCategory.BEHAVIOR)
+    public void ElementsFromPairs(String elements) {
```

Comment:
The format for (pair-based) Data components is as follows:
x1, y1, x2, y2, x3, y3, ..., xn, yn

Should we introduce some sort of specialized format, or is this fine?

Also, is the ElementsFromPairs name representative enough for 2-tuple based Data Series?